### ISSUE TITLE
UDP problem

### ISSUE THREAD

**Issue by timknapen** [arduino/wifishield#28]
Hi, I've been trying to get the WiFiUDPSendReceiveString example to work, but I am not receiving any message on the computer I'm sending to... Also the wifi Shield is not receiving any messages.

is the branch UDP dev still "under construction" or is it supposed to be working already?

(I'm sorry to ask this lame question here, but I can't see where else)

**Comment by jmg22** [comment]
Check issue 23
https://github.com/arduino/wifishield/issues/23

**Comment by timknapen** [comment]
I guess this means that it is supposed to be working. 
Ok, then I'm off debugging again. thanks.

**Comment by timknapen** [comment]
ok, I've kept trying but nothing works.
I use 
- udp_dev branch of the wifi-shield library
- Arduino103 on OSX 
- Arduino UNO 
- WifiShield R3 with firmware updated to the version in the udp_dev branch
  ( at startup it reports firmware 1.0.0 )

I use the wifiUdpSendReceiveString example included in the library to test.
the only change I make is have it connect to our WPA secured network. This works fine with the other examples in the library.
I have tried to communicate using a Udp client and Udp server, both from my computer and from an iPad.
All these apps have been tested an communicate fine with each other.
I have tried using different networks (but all of the WPA encrypted)

I can't receive any UDP messages on the arduino and the other devices don't receive UDP from the arduino.
I've checked and double checked IP's and ports..
I'm out of ideas... anyone?

**Comment by jmg22** [comment]
I think the firmware update has failed , verified this probability

**Comment by timknapen** [comment]
How can you verify if the firmware update succeeded? I always get the confirmitation messages and the FTDI header doesn't give any serial feedback.

**Comment by jimunkim** [comment]
Did not work. Even after upgrade the Wifi Shield firmwares.

I upgraded the Wifi Shield firmware from udp_dev branch (wifi_dnld, wifiHD), and watched confirmation messages both correctly. Set-up all execution environments correctly with the WifiUdpSendReceiveString example included in the library.

Wifi Shield UDP SendReceiveString example still not working. When I send UDP packet from my PC to Arduino Wifi Shield, the serial monitor did not show any message not even once... the receiving PacketSize is 0 all the time.

Is there anybody has an idea to get this work out? any comments will be appreciated...

**Comment by mlafauci** [comment]
we are moving all the recent development on the wifishield to the main 
arduino repository.
in the meantime a fork has been created, where the issue you reported 
was solved.

please have a look at:

https://github.com/mlafauci/Arduino/tree/wifishield-bugfix

Let me know if with the recent binary solve the issue.

Use the binary on the folowing dir:

https://github.com/mlafauci/Arduino/blob/wifishield-bugfix/hardware/arduino/firmwares/wifishield/wifiHD/Release/wifiHD.hex

**Comment by jimunkim** [comment]
I upgraded firmware with wifiHD.elf (41,319KB) from the source above, then the ERROR/LINK, DATA LED is on.
  -> batchisp.exe -device AT32UC3A1256 -hardware usb -operation erase f memory flash wifiHD.elf program verify start reset 0

Or could you let me know how to upgrade firmware with the binary file in the other way?

**Comment by timknapen** [comment]
It doesn't work.
This is step by step what I do:
I downloaded https://github.com/mlafauci/Arduino/tree/wifishield-bugfix and took the WifiHD.hex and WifiHD.elf from  
https://github.com/mlafauci/Arduino/blob/wifishield-bugfix/hardware/arduino/firmwares/wifishield/wifiHD/Release/ and copied them to my Arduino.app (v104) at the path /Applications/Arduino.app/Contents/Resources/Java/hardware/arduino/firmwares/wifishield/
I also copied the files wifi_dnld.elf and wifi_dnld.hex from https://github.com/mlafauci/Arduino/blob/wifishield-bugfix/hardware/arduino/firmwares/wifishield/binary to the same path in my app bundle.

_Note_ that the path ends with /wifishield/ but the update script has this the first line:

```
WIFI_FW_PATH="/hardware/arduino/firmwares/wifi-shield/" 
```

So I change that line in the update script to

```
WIFI_FW_PATH="/hardware/arduino/firmwares/wifishield/"
```

Then I take the wifi shield off from the arduino, add the jumper and plug in the mini USB
Then I open terminal and type 

```
sudo sh /Applications/Arduino.app/Contents/Resources/Java/hardware/arduino/firmwares/wifishield/scripts/ArduinoWifiShield_upgrade.sh -a"/Applications/Arduino.app/Contents/Resources/Java" -f all
```

Then I get this:

```
       Arduino WiFi Shield upgrade
=========================================
Instructions:

To access to the USB devices correctly, the dfu-programmer needs to have the root permissions.

You can upgrade the firmware of the antenna togheter with the shield firmware or only the shield firmware
if there aren't changes on the antenna firmware.

Use the '-h' parameter for help
=========================================

****Upgrade HD WiFi module firmware****

Validating...
167732 bytes used (66.05%)
-n 
Press the RESET button on the shield then type [ENTER] to upgrade the firmware of the shield..


****Upgrade WiFi Shield firmware****

Validating...
257298 bytes used (101.32%)

Done. Remove the J3 jumper and press the RESET button on the shield.
Thank you!
```

Now the DATA, ERROR and LINK leds are on.
I then remove the jumper and press reset.
Those 3 LEDS light up again.

I put the shield back on the Arduino.

I copy the folder https://github.com/mlafauci/Arduino/blob/wifishield-bugfix/libaries/WiFi to /Applications/Arduino.app/Contents/Resources/Java/libraries/
I open the Arduino app and try the example "WifUdpSendReceiveString"
Result: the Serial console tells me
WiFi shield not present

:-(

**Comment by timknapen** [comment]
It puzzles me that you ( mlafauci ) can make this work and I can't. Do you use something else to upload? Do you use another board? Another upload script? another... ???
This is my setup:
Mac OSX 10.8.3
Arduino IDE 1.0.4
Wifi Shield model WIFI R3 with ATMEL 32UC3A1256-U
Arduino UNO

could you maybe also give a description of your setup and how you step-by-step update the firmware?

**Comment by jimunkim** [comment]
F.Y.I. We decided to change plan to using an ethernet shield and a router, instead of the wifi shield we bought. Those worked fine with UDP.

**Comment by timknapen** [comment]
I was also depending on this for a museum installation. The exhibition opened last Friday and the setup had to be replaced by one with 5 XBees and a Mac mini that worked as UDP server. This is very unfortunate, because we have been using Arduino hardware for our installations for a long time now and we love it... I would still like to see this resolved. Is there any way I can help to get this hardware to work properly?

**Comment by mlafauci** [comment]
Try again with the latest binary you find on

https://github.com/mlafauci/Arduino/tree/wifishield-bugfix/hardware/arduino/firmwares/wifishield/wifiHD/Release

Remenmber that it is not yet released and we are testing it.

Il 28/03/2013 07:40, timknapen ha scritto:

**Comment by timknapen** [comment]
Hi Mlafauci, thanks I will test it today.
Could you please describe how you upload and test the firmware? So I can rule out any errors that might occur during the updating of the firmware.
Do you have any way to be sure the firmware update was successful apart from the feedback the shell script gives?

Please tell me anything I can do to debug and test. I feel a bit bad that I can't compile the 32bit stuff myself yet.

best,
tim.

**Comment by mlafauci** [comment]
You can use the shell script. On the binary you tested there was an 
issue on the upgrading binary.
Now it is ok, so you should succeed on the firmware upgrade.

**Comment by timknapen** [comment]
mlafauci... HERO!
it works! :+1:  thanks so much for your work!
I'm starting to make a write-up of every step I did and will post it on the Arduino forum today so everyone can enjoy the wonders of UDP :-)
One thing: I also downloaded the Wifi library from the branch where you told me to take the binary from. It is missing the UDP example, I suggest putting it back in there for other people to test the setup.
